### ISSUE TITLE
Feature request: make hindent work with a *nix shell interpreter written at the top

### ISSUE THREAD

**Issue by k-bx** [chrisdone/hindent#302]
There's a very nice way to write single-hs-file programs in haskell, as shown in [Practical Haskell: Simple File Mirror](https://www.fpcomplete.com/blog/2016/09/practical-haskell-simple-file-mirror-1) series, but it seems hindent doesn't like the `#!/usr/bin/env stack` at the top of an hs file.

Would be very nice to handle that somehow. Thanks!

**Comment by chrisdone** [comment]
Agreed.

**Comment by liff** [comment]
I believe the shebang issue was fixed in #208 but there's still a small annoyance: hindent does not preserve the permissions of the reformatted file.

That is:

```
> echo -e '#!/usr/bin/env stack\n-- stack runghc\nmain = putStrLn "hi"' > example
> chmod +x example
> ./example
hi
> hindent example
> ./example
-bash: ./example: Permission denied
```
